capa/rules.py

```diff
@@ -666,7 +667,11 @@ def rec(statement):
     def _extract_subscope_rules_rec(self, statement):
         if isinstance(statement, ceng.Statement):
             # for each child that is a subscope,
-            for subscope in filter(lambda statement: isinstance(statement, ceng.Subscope), statement.get_children()):
+            for child in statement.get_children():
+                if not isinstance(child, ceng.Subscope):
+                    continue
```

Comment:
mypy can't following the type narrowing in the `filter(isinstance(...))` so we use this alternative construct.